opensafely-core/job-server
# Move ReportPublishRequests to their Report pages

It was confusing for a user to find the ReportPublishRequests when arriving in the staff area with no context other than a report.  This lists them, with their metadata, on the relevant report page, and provides users with approve/reject buttons when a decision hasn't already been made.

#### Testing
1. Grab a copy of the production db and load it locally
2. Pick an analysis request in the staff area, choose one which already has a report
3. Grab the analysis request's slug and generate a report file for it locally with `python manage.py osi_release <slug> <your username>` (the contents of the report is unimportant for this fix)
4. Open the analysis request page the user would see and request it's report is published
5. Go to the report in the staff area (its linked from the analysis request page)
6. Try to approve/reject the request

#### Pending a decision
![](https://p198.p4.n0.cdn.getcloudapp.com/items/yAuJp7Zy/9f90673d-5362-4e02-9b7b-ab17104ceeac.jpg?v=d3d219123b71c635515ef658fe7b71a7)

#### With a decision
![](https://p198.p4.n0.cdn.getcloudapp.com/items/eDurmQ8O/bae39c79-4084-4a86-890c-068ac3ee88ff.jpg?v=6bf89578785b4dc74550c083a9ddbc35)

---

I've also fixed #3153 here since it's a tiny fix but relies on changes in this PR.

Fix: #3150 
Fix: #3153
